### PR TITLE
Add benchmark for image classification

### PR DESCRIPTION
**Pull request by fxmarty**
# What does this PR do?

Add a processing class for image classification, to be usable with autoquantize.

This should be merged only after https://github.com/huggingface/optimum/pull/231 .

Diff: https://github.com/fxmarty/optimum/compare/tests-benchmark...add-benchmark-image-classification

## Before submitting
- [x] Did you write any new necessary tests?

### CODE REVIEW

File: optimum/utils/preprocessing/image_classification.py [hunk 1]
```diff
@@ -0,0 +1,102 @@
...
+from torchvision.transforms import CenterCrop, Compose, Normalize, Resize, ToTensor
+from transformers import FeatureExtractionMixin, ImageClassificationPipeline
+
+from .base import DatasetProcessing
+
+
+class ImageClassificationProcessing(DatasetProcessing):
+    def __init__(self, **kwargs):
+        if "secondary" in kwargs["data_keys"]:
+            raise ValueError("Only one data column is supported for token-classification.")
```
Comment (regisss) [inline optimum/utils/preprocessing/image_classification.py:15]: It should be `image-classification` at the end of the string no?

File: optimum/utils/preprocessing/image_classification.py [hunk 2]
```diff
@@ -0,0 +1,102 @@
...
+        if not isinstance(self.preprocessor, FeatureExtractionMixin):
+            raise ValueError(
+                f"Preprocessor is expected to be a feature extractor, provided {type(self.preprocessor)}."
+            )
+
+    def load_datasets(self):
+        # Downloading and loading a dataset from the hub.
+        raw_datasets = load_dataset(path=self.dataset_path, name=self.dataset_name)
+
+        max_eval_samples = 100  # TODO remove this
```
Comment (regisss) [inline optimum/utils/preprocessing/image_classification.py:30]: To remove I guess

File: optimum/utils/preprocessing/image_classification.py [hunk 3]
```diff
@@ -0,0 +1,102 @@
...
+            pred_label = max(pred, key=lambda x: x["score"])["label"]
+            pred_index = pipeline.model.config.label2id[pred_label]
+            all_preds.append(pred_index)
+
+        return all_labels, all_preds
+
+    def get_metrics(self, predictions: List, references: List, metric: Metric):
+        metrics_res = metric.compute(predictions=predictions, references=references)
+
+        # try to get a good default here
```
Comment (regisss) [inline optimum/utils/preprocessing/image_classification.py:95]: I don't understand this comment

File: tests/benchmark/onnxruntime/test_image_classification.py [hunk 4]
```diff
@@ -0,0 +1,78 @@
...
+            },
+            "metrics": ["accuracy"],
+            "quantization_approach": "dynamic",
+            "operators_to_quantize": ["Add", "MatMul"],
+            "node_exclusion": [],
+            "per_channel": False,
+            "framework": "onnxruntime",
+            "framework_args": {"optimization_level": 1, "opset": 15},
+            "batch_sizes": [8],
+            "input_lengths": [128],
```
Comment (regisss) [inline tests/benchmark/onnxruntime/test_image_classification.py:62]: Is this the size of the images?

**Review by regisss — APPROVED** [review]
It looks great @fxmarty !
I left a few minor comments.

File: optimum/utils/preprocessing/image_classification.py (hunk 2 above)
Comment (fxmarty) [reply optimum/utils/preprocessing/image_classification.py:30]: Thank you! It is removed in the next PR where `max_eval_samples` is provided as an argument.

File: optimum/utils/preprocessing/image_classification.py (hunk 3 above)
Comment (fxmarty) [reply optimum/utils/preprocessing/image_classification.py:95]: Thank you, it wasn't clear. Typically, `metric.compute` may return a number or a dict depending on the metric.

File: tests/benchmark/onnxruntime/test_image_classification.py (hunk 4 above)
Comment (fxmarty) [reply tests/benchmark/onnxruntime/test_image_classification.py:62]: This parameter is ignored, we use `(model.config.image_size, model.config.image_size)` and run the time benchmark only on this size.

File: optimum/utils/preprocessing/image_classification.py (hunk 1 above)
Comment (fxmarty) [reply optimum/utils/preprocessing/image_classification.py]: Yes, fixed, thanks!

**Pull request merged**